experimental/casemapping/src/casemapping.rs

```diff
@@ -114,27 +153,99 @@ impl CaseMapping {
     /// This function only implements simple and common mappings. Full mappings,
     /// which can map one `char` to a string, are not included.
     /// For full mappings, use [`CaseMapping::to_full_uppercase`].
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use icu_casemapping::CaseMapping;
+    ///
+    /// let cm = CaseMapping::new();
+    ///
+    /// assert_eq!(cm.to_uppercase('c'), 'C');
+    /// assert_eq!(cm.to_uppercase('C'), 'C');
+    /// assert_eq!(cm.to_uppercase('ć'), 'Ć');
+    /// assert_eq!(cm.to_uppercase('γ'), 'Γ');
+    ///
+    /// assert_eq!(cm.to_uppercase('ǳ'), 'Ǳ');
+    /// ```
     pub fn to_uppercase(&self, c: char) -> char {
         self.data.get().simple_upper(c)
     }
 
     /// Returns the titlecase mapping of the given `char`.
     /// This function only implements simple and common mappings. Full mappings,
     /// which can map one `char` to a string, are not included.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use icu_casemapping::CaseMapping;
+    ///
+    /// let cm = CaseMapping::new();
+    ///
+    /// assert_eq!(cm.to_titlecase('ǳ'), 'ǲ');
+    ///
+    /// assert_eq!(cm.to_titlecase('c'), 'C');
+    /// assert_eq!(cm.to_titlecase('C'), 'C');
+    /// assert_eq!(cm.to_titlecase('ć'), 'Ć');
+    /// assert_eq!(cm.to_titlecase('γ'), 'Γ');
+    /// ```
     pub fn to_titlecase(&self, c: char) -> char {
         self.data.get().simple_title(c)
     }
 
-    /// Returns the simple case folding mapping of the given char.
+    /// Returns the simple case folding of the given char.
     /// For full mappings, use [`CaseMapping::full_fold`].
+    ///
+    /// You can use the case folding to perform caseless matches on characters
+    /// provided they don't full-casefold to strings. To avoid that situation,
+    /// convert to a string and use [`CaseMapping::full_fold`].
```

Comment:
I am not sure this is understandable or actionable by a normal human being.

Since we added an invariant test for character equivalence I am comfortable with stating it, just noting that there are edge cases in 15.0.

I would also mention the main real application of this function, which is simple case folding, wherever required.

```suggestion
    /// This function can be used to perform caseless matches on
    /// individual characters.
    /// > *Note:* With Unicode 15.0 data, there are three
    /// > pairs of characters for which equivalence under this
    /// > function is inconsistent with equivalence of the
    /// > one-character strings under [`CaseMapping::full_fold`].
    /// > This is resolved in Unicode 15.1 and later.
    ///
    /// For compatibility applications where simple case folding
    /// of strings is required, this function can be applied to
    /// each character of a string.  Note that the resulting
    /// equivalence relation is different from that obtained
    /// by [`CaseMapping::full_fold`]:
    /// The strings "Straße" and "STRASSE" are distinct
    /// under simple case folding, but are equivalent under
    /// default (full) case folding.
```

Reply:
Thanks, that seems much better